### PR TITLE
Added sync, serial-async, and parallel-async examples

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Added sync, serial-async, and parallel-async examples.

Related to:

### CODE REVIEW

File: async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Threading;
+
+namespace ParallelAsyncExample
+{
+    public partial class MainWindow : Window
+    {
+        readonly HttpClient _client = new HttpClient { MaxResponseContentBufferSize = 1_000_000 };
```
Comment (Youssef1313) [inline async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs:14]:
```suggestion
        private readonly HttpClient _client = new HttpClient { MaxResponseContentBufferSize = 1_000_000 };
```

File: async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+        readonly IEnumerable<string> _urlList = new string[]
```
Comment (Youssef1313) [inline async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs:16]:
```suggestion
        private readonly IEnumerable<string> _urlList = new string[]
```

File: async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            "https://docs.microsoft.com/surface",
+            "https://docs.microsoft.com/xamarin",
+            "https://docs.microsoft.com/azure/devops",
+            "https://docs.microsoft.com/system-center",
+            "https://docs.microsoft.com/graph",
+            "https://docs.microsoft.com/education",
+            "https://docs.microsoft.com/gaming",
+        };
+
+        void OnStartButtonClick(object sender, RoutedEventArgs e)
```
Comment (Youssef1313) [inline async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs:39]: Access modifiers are missing to all methods

File: async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            {
+                _resultsTextBox.Text += $"\nControl returned to {nameof(OnStartButtonClick)}.";
+                _startButton.IsEnabled = true;
+            });
+        }
+
+        async Task SumPageSizesAsync()
+        {
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
```
Comment (Youssef1313) [inline async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs:60]: nit: You can use the Stopwatch.StartNew static method.

File: async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+            return byteArray.Length;
+        }
+
+        Task DisplayResultsAsync(string url, byte[] content) =>
+            Dispatcher.BeginInvoke(() =>
+                _resultsTextBox.Text += $"{url,-60} {content.Length,10:#,#}\n")
+                      .Task;
+
+        protected override void OnClosed(EventArgs e) => _client?.Dispose();
```
Comment (Youssef1313) [inline async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs:93]: It doesn't seem possible that `_client` can be null. Is the check for null just defensive?

File: async/async-and-await/cs/readme.md [hunk 6]
```diff
@@ -5,42 +5,35 @@ products:
 - dotnet-core
```
Comment (Youssef1313) [inline async/async-and-await/cs/readme.md:5]: This sample will appear under the ".NET Core" node in the samples browser. Currently, there is no option in the samples browser for .NET 5.0 and later, so nothing to do in the PR. Can you submit an internal request for that?

Here is an image of the current available options under the .NET node.

image

File: async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs (hunk 3 above)
Comment (IEvangelist) [reply async/async-and-await/cs/ParallelAsyncExample/MainWindow.xaml.cs]: They're not missing, they were intentionally omitted. I prefer to omit default access modifiers as I believe it makes it easier to read, and they're redundant. But, I added them.

**Pull request merged**